File: app/src/main/java/com/aware/app/testx/view/RegisterFragment_03.java
package com.aware.app.testx.view;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;

import com.aware.app.testx.R;
import com.aware.app.testx.model.User;
import com.aware.app.testx.presenter.SymptomAdapter;

import java.util.ArrayList;

public class RegisterFragment_03 extends Fragment {

    // views
    private ListView listView;

    // UI list view elements
    public SymptomAdapter adapter;

    // User data class elements
    private User user;

    public RegisterFragment_03(User user) {
        this.user = user;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_register_03, container, false);
        adapter = new SymptomAdapter(getContext());
        listView = view.findViewById(R.id.symptoms_list_view);
        listView.setAdapter(adapter);

        return view;
    }

    public ArrayList<User.Symptom> getSymptoms() {
        ArrayList<User.Symptom> symptoms = new ArrayList<>();

        for (int i=0; i<adapter.getCount(); i++) {
            int checked = adapter.getChecked(i);
            if (checked != -1) {
                String name = adapter.getItem(i)[0].replaceAll(" ", "_").toLowerCase();
                symptoms.add(user.new Symptom(name, checked));
            } else {
                symptoms = null;
                break;
            }
        }

        return symptoms;
    }

}

File: app/src/main/java/com/aware/app/testx/view/RegisterFragment_03.kt
package com.aware.app.testx.view


import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.aware.app.testx.R
import com.aware.app.testx.model.User
import com.aware.app.testx.presenter.SymptomAdapter
import java.util.*
import kotlin.collections.ArrayList

class RegisterFragment_03 : Fragment() {

    private lateinit var adapter : SymptomAdapter

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_register_03, container, false)

        // Set up symptoms list
        adapter = SymptomAdapter(context!!)
        val list = view.findViewById<RecyclerView>(R.id.symptoms_recycler_view)
        list.layoutManager = LinearLayoutManager(context)
        list.adapter = adapter

        return view
    }

    fun getSymptoms(): ArrayList<User.Symptom>? {
        var symptoms: ArrayList<User.Symptom>? = ArrayList()
        for (i in 0 until adapter.itemCount) {
            val checked = adapter.result[i]
            if (checked != -1) {
                val name = adapter.getName(i)
                symptoms?.add(User.Symptom(name, checked))
            } else {
                symptoms = null
                break
            }
        }

        return symptoms
    }
}
